server/src/main/java/com/google/musicanalysis/cache/AnalysisCache.java

```diff
@@ -0,0 +1,86 @@
+package com.google.musicanalysis.cache;
+
+import com.google.musicanalysis.types.*;
+import java.lang.Object;
+import java.lang.ClassNotFoundException;
+import java.time.Instant;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.io.File;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.io.ObjectInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectOutputStream;
+import java.io.IOException;
+
+
+public class AnalysisCache {
+    private HashMap<String, AnalysisGroup> responseMap;
+    
+    public AnalysisCache(){
+        this.responseMap = new HashMap<String, AnalysisGroup>();
+    }
+
+    public void open() {
+        HashMap<String, AnalysisGroup> responseData = null;
+
+        int file = createFile(); // this will have use later.
+
+        try {
+            // .ser for serialized 
+            FileInputStream inFile = new FileInputStream("cachedData.txt");
+            ObjectInputStream inData = new ObjectInputStream(inFile);
+            responseMap = (HashMap<String, AnalysisGroup>) inData.readObject();
+            inData.close();
+            inFile.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return;
+        } catch (ClassNotFoundException c) {
+            c.printStackTrace();
+            return;
+        }
+    }
+
+    public void close() {
+        try {
+            FileOutputStream outFile = new FileOutputStream("cachedData.txt");
+            ObjectOutputStream outData = new ObjectOutputStream(outFile);
+            outData.writeObject(this.responseMap);
+            outData.close();
+            outFile.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void add(String requestUrl, AnalysisGroup responseData) {
```

Comment:
I suggest caching based on the video query string rather than the request url. This will keep unrelated urls out of your cache. E.g., you know the following urls should return the same value:
```
seun.com/api/analysis?video=foo
seun.com/api/analysis?video=foo&lol=get
seun.com/api/analysis?video=foo&rekt=cache
```

Reply:
Will do.